### PR TITLE
SOC-884 Post detail page updates

### PR DESCRIPTION
**Pull request by mklucsarits**
https://wikia-inc.atlassian.net/browse/SOC-884
This pull request is co-dependent on https://github.com/Wikia/mercury/pull/1218. They should be merged at the same time.

@lizlux @garthwebb

### CODE REVIEW

File: front/scripts/main/router.ts [hunk 1]
```diff
@@ -70,7 +70,7 @@ App.Router.map(function () {
 		path: articlePath + 'addPhoto/:title'
 	});
 
-	this.route('discussion', {path: '/d'}, function () {
+	this.route('discussion', {path: '/d/:thing'}, function () {
```
Comment (lizlux) [inline front/scripts/main/router.ts]: What is `thing`?

File: front/styles/main/component/_site-head.scss [hunk 2]
```diff
@@ -65,3 +73,51 @@
...
 	}
 }
+
+hr.theme {
+	border-width: 0;
+	height: 10px;
+	margin: 0;
+}
+
+.books-vertical {
```
Comment (rogatty) [inline front/styles/main/component/_site-head.scss]: You can use a loop to generate these styles, see https://github.com/Wikia/mercury/blob/74eb4c8a197c438bab104cf589cf8838e2a1503c/front/styles/main/component/_smart-banner.scss#L130 for example.

File: front/scripts/main/controllers/DiscussionForumPostController.ts [hunk 3]
```diff
@@ -0,0 +1,9 @@
+/// <reference path="../app.ts" />
+'use strict';
+
+App.DiscussionForumPostController = Em.Controller.extend({
+	init: function () {
+		// Enables vertical-colored theme bar in site-head component
+		this.controllerFor('application').set('themeBar', true);
```
Comment (rogatty) [inline front/scripts/main/controllers/DiscussionForumPostController.ts]: I don't see it being set to `false` anywhere. How about setting it to `true` in the route in [activate hook](http://emberjs.com/api/classes/Ember.Route.html#method_activate) and to `false` in [deactivate hook](http://emberjs.com/api/classes/Ember.Route.html#method_deactivate)?

File: front/scripts/main/components/SiteHeadComponent.ts [hunk 4]
```diff
@@ -5,6 +5,7 @@
 
 App.SiteHeadComponent = Em.Component.extend(App.TrackClickMixin, {
 	classNames: ['site-head'],
+	classNameBindings: ['themeBar'],
```
Comment (rogatty) [inline front/scripts/main/components/SiteHeadComponent.ts]: Please add a default for `themeBar` property.

File: front/scripts/main/controllers/DiscussionForumPostController.ts (hunk 3 above)
Comment (garthwebb) [inline front/scripts/main/controllers/DiscussionForumPostController.ts]: @mklucsarits care to comment?  This must be another merge artifact :/

File: front/scripts/main/router.ts (hunk 1 above)
Comment (garthwebb) [inline front/scripts/main/router.ts]: @mklucsarits ...  and another merge artifact :/  Should this still be here?

Comment (mklucsarits) [inline front/scripts/main/router.ts]: It's the type of page for discussion, which would be 'f' for forum, 'p' for post, or something else. Should it be called 'type' instead?

Comment (lizlux) [inline front/scripts/main/router.ts]: Yeah I think "type" is better than "thing". Wishing there was something better than "type", but that sounds good for now.

**Comment by garthwebb** [comment]
LGTM

**Comment by garthwebb** [comment]
@rogatty I believe @mklucsarits addressed all your feedback so I'm going to merge this.  If you feel there is any additional feedback to give, please feel free to leave it here or open a separate ticket.

**Pull request merged**